cmd/bom/cmd/generate.go

```diff
@@ -343,22 +343,24 @@ func generateBOM(opts *generateOptions) error {
 		return fmt.Errorf("generating doc: %w", err)
 	}
 
-	if opts.outputFile == "" {
-		var renderer serialize.Serializer
-		if opts.format == "json" {
-			renderer = &serialize.JSON{}
-		} else {
-			renderer = &serialize.TagValue{}
-		}
-
-		markup, err := renderer.Serialize(doc)
-		if err != nil {
-			return fmt.Errorf("serializing document: %w", err)
-		}
+	var renderer serialize.Serializer
+	if opts.format == "json" {
+		renderer = &serialize.JSON{}
+	} else {
+		renderer = &serialize.TagValue{}
+	}
 
+	markup, err := renderer.Serialize(doc)
+	if err != nil {
+		return fmt.Errorf("serializing document: %w", err)
+	}
+	if opts.outputFile == "" {
 		fmt.Println(markup)
+	} else {
+		if err := os.WriteFile(opts.outputFile, []byte(markup), 0o264); err != nil {
```

Comment:
What's the reason to use `0o264`?

Reply:
See https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_bom/163/pull-bom-verify/1569541822475145216

Reply:
Yeah but you could use `0o664`, right?

Reply:
See https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_bom/163/pull-bom-verify/1569542654562144256 :)

Reply:
I guess we could skip linter check somehow ?

Reply:
Yep, for example with a comment above like this:
```golang
//nolint:gosec // TODO(gosec): G306: Expect WriteFile
```

Reply:
Fixed